cryptographic_estimators/estimation_renderer.py

```diff
@@ -0,0 +1,115 @@
+from .base_constants import BASE_ALGORITHM, BASE_PARAMETERS, BASE_TIME, BASE_MEMORY, BASE_ADDITIONALO, BASE_QUANTUMO, BASE_TILDEO_ESTIMATE
+from .helper import concat_all_tables, round_or_truncate
+from copy import deepcopy
+from prettytable import PrettyTable
+from sage.rings.all import RR
+
+
+class EstimationRenderer():
+    def __init__(self, show_quantum_complexity=0, show_tilde_o_time=0, show_all_parameters=0, precision=1, truncate=0) -> None:
+        """
+        Creates an estimation renderer
+        INPUT:
+
+        - ``show_quantum_complexity`` -- show quantum time complexity (default: false)
+        - ``show_tilde_o_time`` -- show Ō time complexity (default: false)
+        - ``show_all_parameters`` -- show all optimization parameters (default: false)
+        - ``precision`` -- number of decimal digits output (default: 1)
+        - ``truncate`` -- truncate rather than round the output (default: false)
+        """
+        self._show_quantum_complexity = show_quantum_complexity
+        self._show_tilde_o_time = show_tilde_o_time
+        self._show_all_parameters = show_all_parameters
+        self._precision = precision
+        self._truncate = truncate
+
+    def as_table(self, estimation_result: dict) -> None:
+        """
+        Prints the given estimation dictionary as a table
+        """
+        estimation = deepcopy(estimation_result)
+        if estimation is None:
```

Comment:
as said I think this check should be outside the printing logic. IF it remains here the error thrown would have to be something like "Empty dictionary inputed for printing"